wordpress-mobile/WordPress-iOS
# Add automation to update App Store strings .po file for Jetpack

Adds rough logic to the `update_appstore_strings` lane to update `.po` file for Jetpack as well as WordPress.

cc @AliSoftware. I'm not sure where you're at with the Android side of this, but I hope this will be useful, even though for the Play Store we use the `PlayStoreStrings.po`.

## To test

1. Checkout **a new branch** from this one. `update_appstore_strings` commits and pushes the changes it makes, being on a new branch will prevent it from adding noise to this PR.
1. Optional: Push the branch. If you don't, `update_appstore_strings` will successfully update the `.po` but will fail because it can't push the changes 🤷‍♂️
2. Edit some of the files in `fastlane/appstoreres/jetpack_metadata/`
3. Run `bundle exec fastlane update_appstore_strings version:X`, where `X` is a version value.

You should see a new commit in your history. Looking at it you should see your changes reflected in the `WordPress/Jetpack/Resources/AppStoreStrings.po` file, including the "what's new" version key changing to the version value `X` you specified.

☝️ In your test you might notice a new line in the description, e.g. for WordPress

```diff
 #. translators: Subtitle to be displayed below the application name in the Apple App Store. Limit to 30 characters including spaces and commas!
 msgctxt "app_store_subtitle"
-msgid "Build a website or blog"
+msgid "Build a website or blog
+"
```

That's some kind of bug which I haven't research yet, but is unrelated to these changes.

## Regression Notes
1. Potential unintended areas of impact
N.A.: This does not update any existing automation, it only adds new steps.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

---
